compiler/rustc_middle/src/infer/unify_key.rs

```diff
@@ -11,7 +11,10 @@ pub trait ToType {
 }
 
 #[derive(PartialEq, Copy, Clone, Debug)]
-pub struct UnifiedRegion<'tcx>(pub Option<ty::Region<'tcx>>);
+pub struct UnifiedRegion<'tcx> {
+    pub universe: ty::UniverseIndex,
+    pub value: Option<ty::Region<'tcx>>,
+}
```

Comment:
similar to `TypeVariableValue` and `ConstVariableValue` change this to an enum and only have a universe in the `Unknown` case? :thinking:

Reply:
although it seems redundant, but I thought it might be a good idea to store the unise of universals and existentials in the same place.

Reply:
if region unification is infallible i think we even have to keep the universe even if `value` is `Some`, don't we? :thinking:

Reply:
I don't think so. `universe` field here is redundant if `value` is `Some`.